src/integration/java/io/pinecone/AfterAllTestsListener.java

```diff
@@ -0,0 +1,19 @@
+package io.pinecone;
+
+import io.pinecone.helpers.IndexManagerSingleton;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestPlan;
+
+public class AfterAllTestsListener implements TestExecutionListener {
+
+    @Override
+    public void testPlanExecutionStarted(TestPlan testPlan) {
+        System.out.println("Integration tests starting. Running setup code...");
+
+    }
+    @Override
+    public void testPlanExecutionFinished(TestPlan testPlan) {
+        System.out.println("Integration tests completed. Running cleanup code...");
+        IndexManagerSingleton.getInstance().cleanupResources();
```

Comment:
This is the key detail involving the `TestExecutionListener` for now. Basically, we want to call the `IndexManagerSingleton` cleanup method once at the very end after all tests classes have run, which is what this accomplishes.